components/dashboard/src/settings/PersonalAccessTokens.tsx

```diff
@@ -27,31 +35,218 @@ function PersonalAccessTokens() {
     );
 }
 
+interface EditPATData {
+    name: string;
+    expirationDays: number;
+    expirationDate: Date;
+}
+
+interface PersonalAccessTokenJson {
+    value: string;
+    name: string;
+    expirationTime: string;
+    createdAt: string;
+}
+
+export function PersonalAccessTokenCreateView() {
+    const { enablePersonalAccessTokens } = useContext(FeatureFlagContext);
+
+    const history = useHistory();
+    const [errorMsg, setErrorMsg] = useState("");
+    const [value, setValue] = useState<EditPATData>({
+        name: "",
+        expirationDays: 30,
+        expirationDate: new Date(Date.now() + 30 * 24 * 60 * 60 * 1000),
+    });
+
+    const update = (change: Partial<EditPATData>) => {
+        if (change.expirationDays) {
+            change.expirationDate = new Date(Date.now() + change.expirationDays * 24 * 60 * 60 * 1000);
+        }
+        setErrorMsg("");
+        setValue({ ...value, ...change });
+    };
+
+    const createToken = async () => {
+        if (value.name.length === 0) {
+            setErrorMsg("Token Name must not be empty.");
+            return;
+        }
+        try {
+            const resp = await personalAccessTokensService.createPersonalAccessToken({
+                token: {
+                    name: value.name,
+                    expirationTime: Timestamp.fromDate(value.expirationDate),
+                    scopes: ["*"],
```

Comment:
With https://github.com/gitpod-io/gitpod/pull/14889, this should either be empty, or contain exactly this array `["function:*", "resource:default"]`

Reply:
```suggestion
                    scopes: ["function:*", "resource:default"],
```